### ISSUE TITLE
steam: "Fatal Error: Failed to load steamui.so"

### ISSUE THREAD

**Issue by thypon** [voidlinux/void-packages#4239]
whenever i start steam this is the error message the steamui gives: actually the required 32bit libraries should be installed.

**Comment by necrophcodr** [comment]
After doing a clean install of Steam, and after whatever latest update is out as of writing, the same issue was present for me.

**Comment by Vaelatern** [comment]
I just did the same clean install, removed my alias for steam (which was a wrapper, thus trying to fix it), and now am successfully within Steam.

**Comment by thypon** [comment]
I don't think it's feasible to destroy my 1TB library every time there is a steam update.

**Comment by Vaelatern** [comment]
At least moving it back in later could help. Just needs to go through the install set.

**Comment by Vaelatern** [comment]
Can confirm bug is present on steam reload despite no errors shown.

**Comment by Vaelatern** [comment]
I have two computers, one with steam working, one where it exhibits this error.
On the broken machine, I have just run a `-Su` then that broke steam.
On the working machine, these are the packages that are out of date. https://gist.github.com/83152bf895deb92c958eb780b7ccec0f

**Comment by Vaelatern** [comment]
What's the `nm` command most likely to find the symbols? I tried grepping `common/*` and `xbps-src` for where the `nm` utility is used, but can't find the spot where it decides what libraries are necessary for a shared object file.

**Comment by necrophcodr** [comment]
On my local system I get the following:


[steam.log.txt](https://github.com/voidlinux/void-packages/files/289614/steam.log.txt)

**Comment by necrophcodr** [comment]
That is running with the command line `STEAM_DEBUG=1 STEAM_RUNTIME=0 steam`

With `STEAM_DEBUG=1 STEAM_RUNTIME=1 steam` i get the following:

[steam.log.txt](https://github.com/voidlinux/void-packages/files/289616/steam.log.txt)

**Comment by shaggytwodope** [comment]
This solves the error for me on an Intel HD 4600.

Using the library .so files extracted to a folder (steamlibs in my home folder) downloaded from [here](https://forum.voidlinux.eu/t/odd-networking-issue-with-steam-workaround/237/32).

And running with:

`LD_LIBRARY_PATH=~/steamlibs/ steam`

Completely fixes steam for me. I'm able to run any games I've installed with zero problems. I understand this is a crude work around but hopefully this will help someone.

**Comment by shaggytwodope** [comment]
Contents of the folder:


↘ lstree
./
├─dri/
│ ├─i915_dri.so
│ ├─i965_dri.so
│ ├─kms_swrast_dri.so
│ ├─nouveau_dri.so
│ ├─nouveau_vieux_dri.so
│ ├─r200_dri.so
│ ├─r300_dri.so
│ ├─r600_dri.so
│ ├─radeon_dri.so
│ ├─radeonsi_dri.so
│ ├─swrast_dri.so
│ └─vmwgfx_dri.so
├─libGL.so.1@ → libGL.so.1.2.0
└─libGL.so.1.2.0

**Comment by Pieniasty** [comment]
I can confirm, shaggytwodope's method works for me, it fixed Steam completly.

**Comment by Maks97** [comment]
Unfortunately it didn't fix it for me though it was giving me different errors this time around (unable to load the drivers, driver pointers missing..Progress, I guess?)   
```
libGL error: unable to load driver: r600_dri.so
libGL error: driver pointer missing
libGL error: failed to load driver: r600
libGL error: unable to load driver: r600_dri.so
libGL error: driver pointer missing
libGL error: failed to load driver: r600
libGL error: unable to load driver: swrast_dri.so
libGL error: failed to load driver: swrast
```

(For the record I have an AMD card, Radeon HD 6850 specifically. Not sure if that has anything to do with it though.)

**Comment by Pieniasty** [comment]
It looks like it doesn't see the drivers in the new location, did you download them, unpack to some directory and started steam with LD_LIBRARY_PATH pointed to it?

**Comment by Maks97** [comment]
Yes (I did `export` the LD_LIBRARY_PATH but same thing happens either way), and just in case I really FUBAR'd, I reinstalled Steam, to no avail.

Running it with `STEAM_DEBUG=verbose` doesn't really yield much useful information either. (not that I was expecting anything)

**Comment by nonchip** [comment]
happens for me too, see https://nonchip.de/paste/113 (last line says "self updating starts", reused the paste, sorry for the inconvenience) and https://nonchip.de/paste/114 for logs.
installed `steam-1.0.0.52_4 mesa-intel-dri-32bit-12.0.1_1 libGL-32bit-12.0.1_1` like 5 minutes ago, removed `~/.{local/share/S,s}team*` beforehand.

`ldd steamui.so` (with paths set like steam does) does not show any missing deps, but ldd'ing the whole steam folder shows a few: https://www.nonchip.de/paste/115

if there's anything I can do to help resolve this issue, please feel free to just ask/tell me, I'd love to a) help and b) see steam running on void :)

**Comment by nonchip** [comment]
also valve's own log WITH debugging enabled doesn't show ANYthing useful, of course -_-

[steamlog.txt](https://github.com/voidlinux/void-packages/files/422874/steamlog.txt)

**Comment by nonchip** [comment]
I just `xlocate`d the libs missing in https://www.nonchip.de/paste/115, found most of those, and installed `libavcodec libavformat libavresample libavutil icu-libs libswresample libswscale libv8` (+ their 32bit versions) although most of them seem webkit dependencies and should be brought along w/ steam's internal browser, but you can't be sure...

still doesn't help though. "failed to load steamui.so" without any useful log.

**Comment by nonchip** [comment]
this file btw seems to be the strace of `steamui.so` getting loaded JUST FINE, wtf is valve fucking up here again? 0.o
[steam.strace.27983.txt](https://github.com/voidlinux/void-packages/files/424705/steam.strace.27983.txt)

**Comment by whitephoenix0** [comment]
Just to add some clarified information, the libs people are using to fix this are from debian, downloaded via http://crct.de/debianlibgl.tar.bz2. Here's the folder structure of that archive:

`libGL.so.1.2.0`
`libGL.so.1`
`dri/`

the "dri" folder contains the following:

`i915_dri.so`
`i965_dri.so`
`kms_swrast_dri.so`
`nouveau_dri.so`
`nouveau_vieux_dri.so`
`r200_dri.so`
`r300_dri.so`
`r600_dri.so`
`radeon_dri.so`
`radeonsi_dri.so`
`swrast_dri.so`
`vmwgfx_dri.so`

**Comment by nonchip** [comment]
@whitephoenix0 yes, as i said above. "shaggytwodope's fix". which doesn't work on my i915 based system. on my geforce based system it's not even required, there i just run steam as installed by xbps.

**Comment by aeosynth** [comment]
can void either make the steam package depend on all the required 32 bit libraries, or create a `steam-runtime` meta package which bundles all of them? installing steam on void takes entirely too much work.

**Comment by iamgreaser** [comment]
Void amd64 here.

Using the Debian libGL files workaround *and* removing the i965_dri.so file from fixes this specific issue for me *but* I cannot log in. In other words, #4167 is most likely still valid.

STEAM_RUNTIME=0 does make the steamui.so error crop up again so I may still be missing libs here.
